### DISCUSSION TITLE
typing.Unpack warning

### DISCUSSION THREAD

**Discussion by anagorko** [pydantic/pydantic#9241] [Question] [answered]
I'm at a loss trying to fix the following pydantic warning
```
.venv/lib/python3.11/site-packages/pydantic/_internal/_generate_schema.py:386: UserWarning: typing.Unpack is not a Python type (it may be an instance of an object), Pydantic will allow any object with no validation since we cannot even enforce that the input is an instance of the given type. To get rid of this error wrap the type with `pydantic.SkipValidation`.
  warn(
```

when running the following code (a MWE extracted from my use-case):
```
import pydantic

class X(pydantic.BaseModel):
    model_config = pydantic.ConfigDict(arbitrary_types_allowed=True)

    x: tuple[str, *tuple[int, ...]]
```

Its not just a warning:
```
X(x=("b", 2, 3))
```
fails with an error message
```
pydantic_core._pydantic_core.ValidationError: 1 validation error for X
x
  Tuple should have at most 2 items after validation, not 3 [type=too_long, input_value=('b', 2, 3), input_type=tuple]
```

Note that
```
y: tuple[str, *tuple[int, ...]] = ("a", 1)
z: tuple[str, *tuple[int, ...]] = ("b", 2, 3)
```
works fine: mypy doesn't raise any errors.

Any help would be appreciated.

**Comment by Viicos** [comment, marked as answer]
I'm not sure tuple unpacking is supported yet, but to be confirmed. This would be a nice feature, although might require some work

**Reply by anagorko** [reply]
Thank you, indeed it seems that this is not supported yet.

**Reply by samuelcolvin** [reply]
Ye, it's not yet supported, please create a feature request issue.

Of course, if you really want to dig in, a pull request would be great, but as @Viicos says, it might be quite a lot of work.
